### PR TITLE
Design system v2 buttons implemented in JSX

### PR DESCRIPTION
**Pull request by affordances**
Screen Shot 2022-01-14 at 12 28 36 PM

### CODE REVIEW

**Review by zakhap — APPROVED** [review]
nice

**Review by zakhap — CHANGES_REQUESTED** [review]
ok, still gotta figure out why the imports are wacky

File: client/scripts/views/components/cw_button/cw_button.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { ComponentType, StyleProps } from '../component_kit/types';
+import { getClasses } from '../component_kit/helpers';
+
+export const CWButton: m.Component<
+  {
+    styleProps: StyleProps;
+    label: string;
+    onclick: (e?: MouseEvent) => void;
+  },
+  Record<string, unknown>
```
Comment (zakhap) [inline client/scripts/views/components/cw_button/cw_button.tsx:15]: can you explain this Record object as the state object?

Comment (affordances) [reply client/scripts/views/components/cw_button/cw_button.tsx:15]: TS didn't like the empty object and recommended that instead. We should probably settle on an official type for Mithril component state if there isn't one already.

File: client/scripts/views/components/component_kit/types.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  Button = 'Button',
+  GradientButton = 'GradientButton',
+}
+
+export enum ButtonType {
+  Primary = 'primary',
+  Secondary = 'secondary',
+}
+
+export type StyleProps = {
```
Comment (zakhap) [inline client/scripts/views/components/component_kit/types.ts:11]: Should be clarified as`ButtonStyleProps` as I assume there will be many different types of StyleProps defined, for each class of components

**Review by zakhap — APPROVED** [review]
This all looks good to me!

**Comment by zakhap** [comment]
Gonna wait a few to merge–first gotta deploy to staging and make sure the babel jsx transformers don't blow up on production

**Review by zakhap — CHANGES_REQUESTED** [review]
somehow reverts all new sidebar changes

**Review by zakhap — APPROVED** [review]
Works for me!

**Pull request merged**